src/Analyzers/Core/Analyzers/RemoveUnusedParametersAndValues/AbstractRemoveUnusedParametersAndValuesDiagnosticAnalyzer.SymbolStartAnalyzer.cs

```diff
@@ -265,16 +266,49 @@ private bool IsUnusedParameterCandidate(IParameterSymbol parameter, Cancellation
                 return false;
             }
 
-            // Don't report on valid GetInstance method of ICustomMarshaler.
-            // See https://docs.microsoft.com/dotnet/api/system.runtime.interopservices.icustommarshaler#implementing-the-getinstance-method
-            if (method is { MetadataName: "GetInstance", IsStatic: true, Parameters.Length: 1, ContainingType: { } containingType } methodSymbol &&
-                methodSymbol.Parameters[0].Type.SpecialType == SpecialType.System_String &&
-                containingType.AllInterfaces.Any((@interface, marshaler) => @interface.Equals(marshaler), _iCustomMarshaler))
+            if (method.ContainingType is { } containingType)
             {
-                return false;
+                // Don't report on valid GetInstance method of ICustomMarshaler.
+                // See https://docs.microsoft.com/dotnet/api/system.runtime.interopservices.icustommarshaler#implementing-the-getinstance-method
+                if (method is { MetadataName: "GetInstance", IsStatic: true, Parameters: [{ Type.SpecialType: SpecialType.System_String }] } &&
+                    containingType.AllInterfaces.Any((@interface, marshaler) => @interface.Equals(marshaler), _iCustomMarshaler))
+                {
+                    return false;
+                }
+
+                // 2 first `int` parameters of an interpolated string handler
+                // constructor are mandatory. Therefore, do not report them as unused
+                if (IsInterpolatedStringHandlerMandatoryConstructorParameter(parameter, method, containingType, interpolatedStringHandlerAttributeType))
+                {
+                    return false;
+                }
             }
 
             return true;
+
+            static bool IsInterpolatedStringHandlerMandatoryConstructorParameter(
+                IParameterSymbol parameter,
+                IMethodSymbol method,
+                ITypeSymbol containingType,
+                INamedTypeSymbol? interpolatedStringHandlerAttributeType)
+            {
+                if (parameter.Type.SpecialType != SpecialType.System_Int32)
+                    return false;
+
+                if (method.MethodKind != MethodKind.Constructor)
+                    return false;
+
+                if (!((method.Parameters is [var firstParameter, ..] && firstParameter == parameter) ||
+                    (method.Parameters is [_, var secondParameter, ..] && secondParameter == parameter)))
```

Comment:
this clause confuses me.  shouldn't htat be an `&&` not an `||`?

Reply:
This condition is checking whether the given parameter is the first _or_ the second parameter of a method. If it is false, we early return

Reply:
gotcha.